File: 2800-minimum-string-length-after-removing-substrings/minimum-string-length-after-removing-substrings.java
class Solution {
    public int minLength(String s) {
        Stack<Character> stack = new Stack<>();
        for (int i = 0; i < s.length(); i++) {
            char current = s.charAt(i);
            if (stack.isEmpty()) {
                stack.push(current);
                continue;
            }
            
            if (current == 'B' && stack.peek() == 'A') {
                stack.pop();
            } else if (current == 'D' && stack.peek() == 'C') {
                stack.pop();
            } else {
                stack.push(current);
            }
        }

        return stack.size();
    }
}

File: 2800-minimum-string-length-after-removing-substrings/minimum-string-length-after-removing-substrings.cpp
class Solution {
public:
    int minLength(string s) {
        char stack[s.length() + 1];
        int top = -1;

        for (int i = 0; i < s.length(); i++) {
            char current = s[i];
            if (top > -1 && ((current == 'B' && stack[top] == 'A') || (current == 'D' && stack[top] == 'C'))) {
                top--;
            } else {
                stack[++top] = current;
            }
        }
        return top + 1;
    }
};